lms/static/scripts/frontend_apps/components/test/SubmitGradeForm-test.js

```diff
@@ -247,23 +213,47 @@ describe('SubmitGradeForm', () => {
 
     it('closes the spinner after the grade has posted', async () => {
       const wrapper = renderForm();
-      wrapper.find('button[type="submit"]').simulate('click');
+
+      await waitForGradeFetch(wrapper);
+      wrapper.find('form').simulate('submit');
 
       await waitFor(() => {
         wrapper.update();
         return !wrapper.exists('FullScreenSpinner');
       });
     });
 
-    it('calls grading service', () => {
+    [
+      { gradeValue: '10', expectedGrade: 1 },
+      { gradeValue: '5.5', expectedGrade: 0.55 },
+      { gradeValue: '2', expectedGrade: 0.2 },
+    ].forEach(({ gradeValue, expectedGrade }) => {
+      it('calls grading service with right grade', () => {
+        const wrapper = renderForm();
+
+        wrapper.find(inputSelector).getDOMNode().value = gradeValue;
+        wrapper.find('form').simulate('submit');
+
+        assert.calledWith(fakeGradingService.submitGrade, {
+          student: fakeStudent,
+          grade: expectedGrade,
+          comment: undefined,
+        });
+      });
+    });
+
+    it('throws if the value is not a number', async () => {
       const wrapper = renderForm();
-      wrapper.find('button[type="submit"]').simulate('click');
 
-      assert.calledWith(fakeGradingService.submitGrade, {
-        student: fakeStudent,
-        grade: 1,
-        comment: undefined,
-      });
+      try {
+        await wrapper
+          .find('form')
+          .props()
+          .onSubmit({ preventDefault: sinon.stub() });
+        assert.fail('Error was not thrown');
```

Comment:
Huh. I wasn't aware of `assert.fail` despite how obvious it is in the Chai docs.

Reply:
If I'm honest, I'm not super happy with how this test ended up looking.

It's true that `assert.fail` more or less shows a clear intent, but it does throws an error that ends up being caught by the `catch` block. Since the message does not match, the test ends up failing as intended, but only incidentally.

My ideal solution would be having something like `assert.rejects(() => onSubmit(...))`, but I couldn't find anything like that in either sinon or chai docs.

I even tried to make it work with `assert.throws(...)`, but it does not work with asynchronous callbacks (which makes sense).

Reply:
We do have an `assert.rejects` implementation in the client, but the most common pattern across our apps for this is actually:

```js
let error;
try {
  await doSomething();
} catch (e) {
  error = e;
}
assert.equal(error, ...)
```

Reply:
I think I'll go that way. At least the error will be more explicit.